src/core/socket-bind.h

```diff
@@ -0,0 +1,22 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+#pragma once
+
+#include "unit.h"
+
+#if HAVE_LIBBPF
+struct bpf_link;
+struct SocketBind {
+        struct bpf_link *ipv4_link;
+        struct bpf_link *ipv6_link;
+};
+#else
+struct SocketBind {
+};
+#endif
```

Comment:
This doesn't look right at all. In the !HAVE_BPF case, we'd be creating a zero-size structure (but allocating something for it since we can't allocate 0 bytes), and still paying the cost of a pointer. The abstraction must be "zero cost": if compiled out, there should be exactly zero code or memory wasted. (And less because of the tiny memory waste, but more because of the cognitive overhead of having in-memory structures that serve no purpose.)

But also in the HAVE_BPF case, we don't want to allocate a tiny memory structure just to make an indirection.

I think the easiest way to handle this would be to do
```C
struct Unit {
...
#if HAVE_LIBBPF
        struct bpf_link *ipv4_link;
        struct bpf_link *ipv6_link;
#endif 
...
} Unit;
```